libevmasm/SimplificationRules.cpp

```diff
@@ -82,6 +85,7 @@ Rules::Rules()
 	Y.setMatchGroup(5, m_matchGroups);
 
 	addRules(simplificationRuleList(A, B, C, X, Y));
+	assertThrow(!m_rules[byte(Instruction::ADD)].empty(), OptimizerException, "Rule list not properly initialized.");
```

Comment:
Yeah, well - these assertions won't help - no "sane" code path will lead here with empty rules and in the error case execution will abort/throw/whatever before ever reaching this... but it doesn't hurt either I gues.

Reply:
Ah ok - in a potential second run with the same, resp. a failed static initialization, this will still throw. That makes sense.

Reply:
I'm still not so sure. Perhaps the test should rather be after the declaration of the static variables. I think I'll add another check just to be sure.